OneCommunityGlobal/HGNRest
# Abdel add home country option to user setup

# Description
a.	(PRIORITY MEDIUM) Jae: Later PR/AFTER ABOVE: Create option for “home country”(WIP Abdel)
i.	When people fill out their own Onboarding details (see PR [1073](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1073)), I’d like them to have an option to represent their home country if it is different from the country they are currently residing in. If they choose that option, it should update the map with their details for their country option instead of their location input for timezone purposes. 
ii.	Field: “Do you represent a country other than your current residence?”
iii.	“i”: One Community is a global effort and international team that has had volunteers volunteering from and/or representing <# (currently 60)> countries around the world. Complete this field if you are currently residing in a country other than your own and wish to be represented on our global contributors <link it> map with your birth country instead of your current country or residence.
iv.	This option also needs a “Read the requirements” box AND a followup confirmation popup:
1.	Box mouseover of the “Read the requirements” text: Please only choose this option if you were born and lived for a fair amount of time (10+ years) in the country you are stating you represent.
a.	Please make this editable by Owners using the already existing code for this 
2.	Popup: Thank you for representing our global community! By stating you are a native of a country other than the one you are currently living in, you certify you were born and/or lived the majority of your life and call it “home”.
a.	Yes option: input is accepted and it says: Thank you!
b.	No option: input is cleared and it says: No problem, thank you! 
3.	Please add the popup field in # 2 to the Other Links>Popup Management too
v.	ALSO: Request they add a photo, see below.

## Related PRS (if any):
This backend PR is related to the [#1608](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1608) frontend PR.

## How to test:
1. check into current branch
2. set up the env variables as follow:

To get the client ID, client secret and refresh token. Please refer to the video below:

https://www.youtube.com/watch?v=-rcRf7yswfM&t=10s&ab_channel=MafiaCodes

To get free TIMEZONE API key: Go to https://opencagedata.com/pricing



```
REACT_APP_EMAIL =<your_test_email>
REACT_APP_EMAIL_CLIENT_ID =<follow the video for instructions>
REACT_APP_EMAIL_CLIENT_SECRET =<follow the video for instructions>
REACT_APP_EMAIL_REFRESH_TOKEN =<follow the video for instructions>
REACT_APP_EMAIL_CLIENT_REDIRECT_URI = https://developers.google.com/oauthplayground
sendEmail = TRUE
TIMEZONE_PREMIUM_KEY=<API Key>
TIMEZONE_COMMON_KEY=<API Key>

```


2. do `npm run build`
3. do `npm start`
